core/store/ChainStore/ChainStore.go

```diff
@@ -1279,6 +1369,45 @@ func (bd *ChainStore) GetAccount(programHash Uint160) (*account.AccountState, er
 	return accountState, nil
 }
 
+func (bd *ChainStore) IsStateUpdaterVaild(Tx *tx.Transaction) bool {
+	su := Tx.Payload.(*payload.StateUpdate)
+
+	stateUpdater, err := bd.GetStateUpdater()
+	if err != nil {
+		return false
+	}
+
+	findflag := false
+	for k := 0; k < len(stateUpdater); k++ {
+		log.Trace(fmt.Sprintf("StateUpdate updaterPublickey %d: %x %x", k, stateUpdater[k].X, stateUpdater[k].Y))
+
+		if su.Updater.X.Cmp(stateUpdater[k].X) == 0 && su.Updater.Y.Cmp(stateUpdater[k].Y) == 0 {
+			findflag = true
+			break
+		}
+	}
+
+	if !findflag {
+		return false
+	}
+
+	return true
+}
+
+func (bd *ChainStore) GetState(namespace []byte, key []byte) ([]byte, error) {
+
+	// stateKey
+	statePrefix := []byte{byte(ST_STATES)}
+	stateKey := append(statePrefix, namespace...)
+	stateKey = append(stateKey, key...)
```

Comment:
this desgin may have problem since we can not figure out namespace and key from the stateKey. 
case: namespace: `dna`, key: `node3`,  and namespace:`dnanode`, key:`3` will share the same underlying data。
recommendation: Use len(namespace) + namespace + key schema to construct the stateKey